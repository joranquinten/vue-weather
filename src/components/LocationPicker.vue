<template>
  <div class="location-picker">
    <fieldset>
      <legend>For location</legend>
      <label for="latitude">Latitude</label>
      <input type="number" id="latitude" v-model="location.latitude" />
      <label for="longitude">Longitude</label>
      <input type="number" id="longitude" v-model="location.longitude" />
    </fieldset>
    <button @click="updateAction">Get weather for this position</button>
  </div>
</template>

<script>
export default {
  name: 'LocationPicker',
  props: {
    location: {
      type: Object,
      default: () => ({ latitude: null, longitude: null })
    }
  },
  methods: {
    updateAction: function() {
      this.$emit('location-change', this.location)
    }
  }
}
</script>

<style scoped></style>
