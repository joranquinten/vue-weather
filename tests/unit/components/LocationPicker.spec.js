import { shallowMount, mount } from '@vue/test-utils'
import LocationPicker from '../../../src/components/LocationPicker'

describe('LocationPicker.vue', () => {
  it('renders without crashing', () => {
    const wrapper = shallowMount(LocationPicker)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders props.location when passed', () => {
    const wrapper = shallowMount(LocationPicker, {
      propsData: {
        location: {
          latitude: 1,
          longitude: 2
        }
      }
    })
    expect(wrapper.props().location.latitude).toBe(1)
    expect(wrapper.props().location.longitude).toBe(2)

    expect(wrapper).toMatchSnapshot()
  })

  it('should emit the locationChange event on update', () => {
    const wrapper = mount(LocationPicker, {
      propsData: {
        location: {
          latitude: 1,
          longitude: 2
        }
      }
    })

    wrapper.vm.updateAction()

    expect(wrapper.emitted()['location-change'][0]).toEqual([
      { latitude: 1, longitude: 2 }
    ])
  })
})
