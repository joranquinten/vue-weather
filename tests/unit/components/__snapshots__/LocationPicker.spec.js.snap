// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPicker.vue renders props.location when passed 1`] = `
<div class="location-picker">
  <fieldset>
    <legend>For location</legend> <label for="latitude">Latitude</label> <input type="number" id="latitude"> <label for="longitude">Longitude</label> <input type="number" id="longitude">
  </fieldset> <button>Get weather for this position</button>
</div>
`;

exports[`LocationPicker.vue renders without crashing 1`] = `
<div class="location-picker">
  <fieldset>
    <legend>For location</legend> <label for="latitude">Latitude</label> <input type="number" id="latitude"> <label for="longitude">Longitude</label> <input type="number" id="longitude">
  </fieldset> <button>Get weather for this position</button>
</div>
`;
